Framework/Geometry/src/Instrument/DetectorInfo.cpp

```diff
@@ -158,7 +159,13 @@ double DetectorInfo::twoTheta(const size_t index) const {
   }
 
   const auto sampleDetVec = position(index) - samplePos;
-  return sampleDetVec.angle(beamLine);
+  try {
+    return sampleDetVec.angle(beamLine);
+  } catch (const std::runtime_error &e) {
```

Comment:
I'm still a bit uncomfortable with us catching this at such a low level. The other parts of this function still throw if there are errors. A user of this function would now have to check for both an exception and a `NaN` to confirm there was no error and it would pass silently.

Is the error that Polaris are seeing just with`ConvertUnits`? Can we just catch the errors there and handle them? Other users of this code may want to handle this situation differently and would then just have to catch any exceptions that indicate an error and handle it accordingly.

Reply:
I've moved the check into convert units. If we need to get twotheta from polaris we will need to have these checks.